Python/Product/Profiling/Profiling/UserInputService.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+namespace Microsoft.PythonTools.Profiling {
+    using System;
+    using System.ComponentModel.Composition;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Windows;
+    using Microsoft.PythonTools.Infrastructure;
+    using Microsoft.PythonTools.Interpreter;
+
+    /// <summary>
+    /// Implements a service to collect user input for profiling and generate a <see cref="TargetCommand"/>.
+    /// </summary>
+    [Export(typeof(IUserInputService))]
+    class UserInputService : IUserInputService {
+        private readonly CommandBuilder _commandBuilder;
+        private readonly UserDialog _userDialog;
+
+        [ImportingConstructor]
```

Comment:
I was under the impression that you only had to use `ImportingConstructor` when your constructor is importing MEF types that are exported by someone else. I could be wrong here though, I'm not a MEF expert.

Reply:
https://learn.microsoft.com/en-us/dotnet/framework/mef/attributed-programming-model-overview-mef#:~:text=To%20fill%20the,declare%20an%20import.

Yeah I think you're right, it's used when there're parameters in the class's constructor that need to be imported(inject dependencies through the constructor), since mine doesn't have any parameters, it should be removed. I will run some tests without this attribute to make sure it still works.